efi/preinstall/check_pcr7.go

```diff
@@ -0,0 +1,537 @@
+				// sure it's consistent with the measured digest.
+				expectedDigest := tcglog.ComputeEFIVariableDataDigest(pcrAlg.GetHash(), data.UnicodeName, data.VariableName, data.VariableData)
+				if !bytes.Equal(ev.Digests[pcrAlg], expectedDigest) {
+					return nil, fmt.Errorf("event data inconsistent with measured digest for EV_EFI_VARIABLE_DRIVER_CONFIG event (name:%q, GUID:%v, expected digest:%#x, measured digest:%#x)",
+						data.UnicodeName, data.VariableName, expectedDigest, ev.Digests[pcrAlg])
+				}
+
+				sigDb, err := checkSecureBootVariableData(data)
+				if err != nil {
+					return nil, err
+				}
+				if data.UnicodeName == "db" {
+					// Capture the db from the log for future use.
+					// We don't check the EFI_SIGNATURE_LIST types contained in db. Any OS component with a valid
+					// Authenticode signature (WIN_CERT_TYPE_PKCS_SIGNED_DATA) or a valid PKCS7 signature
+					// (WIN_CERT_TYPE_EFI_GUID with the type EFI_CERT_TYPE_PKCS7_GUID) is authenticated with
+					// signature verification using the matching EFI_CERT_X509_GUID entry first, and the profile
+					// generation in secboot efi makes sure a signed binary has a matching root in the relevant
+					// trust stores. The presence of digest signatures (EFI_CERT_SHA*) shouldn't matter because
+					// these are only used to authenticate unsigned images (which the profile generation in the
+					// secboot efi package rejects) or as a fallback for signed image where signature verification
+					// fails. Digests may be permitted for authenticating unsigned pre-OS components.
+					db = sigDb
+				}
+			case tcglog.EventTypeEFIAction:
+				// This branch exists here for documentation purposes - it falls through to the
+				// default branch below, which returns an error.
+				//
+				// An EV_EFI_ACTION events with the string "UEFI Debug Mode" appears at the
+				// start of the log if a debugging endpoint is enabled. It's also possible that
+				// EV_EFI_ACTION events are used for other conditions in PCR7 that weaken device
+				// security (eg, the "DMA Protection Disabled" event).
+				//
+				// In general, it's not normal to see EV_EFI_ACTION events and these indicate some
+				// sort of abnormal condition that has a detrimental effect on device security.
+				// WithSecureBootPolicyProfile() will generate an invalid policy in this case because
+				// it doesn't emit them.
+				//
+				// Just return an error here to prevent the use of WithSecureBootPolicyProfile(). The
+				// "UEFI Debug Mode" and "DMA Protection Disabled" cases are already picked up by the
+				// firmware protection checks, so we don't need any special handling here.
+				fallthrough
+			default:
+				// Anything that isn't EV_EFI_VARIABLE_DRIVER_CONFIG ends up here.
+				return nil, fmt.Errorf("unexpected %v event %q whilst measuring config", ev.EventType, ev.Data)
+			}
+		case tcglogPhasePreOSAfterMeasureSecureBootConfig, tcglogPhaseOSPresent:
```

Comment:
once we have more helpers maybe we can explore whether splitting in two cases with a little bit of repetition is easier to follow than the code below the rechecks the phase here and there

Reply:
I've done this now. I hope it's a little bit cleaner.